csharp/FigiApiCsharpExample/OpenFIGIRequest.cs

```diff
@@ -0,0 +1,115 @@
+        private static readonly string[] YellowKeys = { "Govt", "Corp", "Mtge", "M-Mkt", "Muni", "Pfd", "Equity", "Comdty", "Curncy", "Index" };
+
+        private static readonly Regex Isin = new Regex("[a-zA-Z]{2}\\w{10}");
+        private static readonly Regex Figi = new Regex("BBG\\w{9}");
+
+        public OpenFIGIRequest(string value)
+        {
+            marketSecDes = GetMarketSector(value) ?? string.Empty;
+            exchCode = GetExchangeCode(value) ?? string.Empty;
+            if (!string.IsNullOrEmpty(marketSecDes))
+                value = value.Replace(marketSecDes, string.Empty);
+            if (!string.IsNullOrEmpty(exchCode))
+                value = value.Replace(exchCode, string.Empty);
+            if (!string.IsNullOrEmpty(value))
+                idValue = string.Join(" ", value.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries));
+
+
+            idType = "TICKER";
+            if (!string.IsNullOrEmpty(idValue))
+            {
+                if (Figi.IsMatch(idValue))
+                    idType = "ID_BB_GLOBAL";
+                else if (Isin.IsMatch(idValue))
+                    idType = "ID_ISIN";
+            }
+        }
+
+        private OpenFIGIRequest()
+        {
+            this.exchCode = this.marketSecDes = this.currency = string.Empty;
+        }
+
+        public OpenFIGIRequest(string idType, string idValue)
+            : this()
+        {
+            this.idType = idType;
+            this.idValue = idValue;
+        }
+
+        public OpenFIGIRequest WithExchangeCode(string exchCode)
+        {
+            this.exchCode = exchCode;
+            //micCode = string.Empty; ;
+            return this;
+        }
+
+        public OpenFIGIRequest WithMicCode(string micCode)
+        {
+            //this.micCode = micCode;
+            exchCode = string.Empty;
+            return this;
+        }
+
+        public OpenFIGIRequest WithCurrency(string currency)
+        {
+            this.currency = currency;
+            return this;
+        }
+
+        public OpenFIGIRequest WithMarketSectorDescription(string marketSectorDescription)
+        {
+            marketSecDes = marketSectorDescription;
+            return this;
+        }
+
+        [JsonProperty("idType")]
+        [XmlElement("idType")]
+        public string idType { get; set; }
+        
+        [JsonProperty("idValue")]
+        [XmlElement("idValue")]
+        public string idValue { get; set; }
+        
+        [JsonProperty("exchCode")]
+        [XmlElement("exchCode")]
+        public string exchCode { get; set; }
+
+        //[JsonProperty("micCode")]
```

Comment:
Why is this one commented?

Reply:
Sending this
```
[{"idType":"ID_ISIN","idValue":"US4592001014","exchCode":"","currency":"","marketSecDes":""}]
```

Or this
```
[{"idType":"ID_ISIN","idValue":"US4592001014","exchCode":null,"micCode":null,"currency":"","marketSecDes":""}]
```

was returning error for having exchCode and micCode.
This comment is temporary.

Now you already answered about this problem.

Thanks!